[camera]remove QueueHelper class

hellohuanlin opened the pull request:
Removes QueueHelper class and use function instead. 

Also renamed file to `QueueUtils` to be consistent with `CameraTestUtils`

*List which issues are fixed by this PR. You must list at least one issue.*

NA

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

No version change: just minor code cleanup. 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

jmagman commented on packages/camera/camera/ios/Classes/QueueUtils.m:
```diff
@@ -2,18 +2,14 @@
-@implementation QueueHelper
-
-+ (void)ensureToRunOnMainQueue:(void (^)(void))block {
+void FLTEnsureToRunOnMainQueue(dispatch_block_t block) {
```
Frankly I still don't think these little utility function or classes should exist, it's just a wrapper around 3 lines of well-understood Foundation code, your Objective-C eye can just skim over some code about `NSThread` instead of needing to understand what `FLTEnsureToRunOnMainQueue` does.  In my opinion it would be clearer to just check if you're on the main thread in the places you need to.  And then you have access to `_cmd` 
https://github.com/flutter/plugins/blob/10a59af20f143d52d5a2c471bb9e7e3e79367a1e/packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m#L349

hellohuanlin replied:
Some params are not objects and it's annoying to wrap/unwrap them, so I will not use `performSelector` approach.

cyanglaz reviewed (commented):
What was the motivation to do this? I thought the existing code was cleaner?

hellohuanlin commented:
> What was the motivation to do this? I thought the existing code was cleaner?

@cyanglaz I changed from objc static method to C functions and then @jmagman recommended to remove this helper but it didn't turn out as good, so i just reverted it back.

hellohuanlin commented on packages/camera/camera/ios/Classes/FLTThreadSafeEventChannel.m:
```diff
@@ -21,10 +21,15 @@ - (instancetype)initWithEventChannel:(FlutterEventChannel *)channel {
 - (void)setStreamHandler:(NSObject<FlutterStreamHandler> *)handler
               completion:(void (^)(void))completion {
-  FLTEnsureToRunOnMainQueue(^{
+  dispatch_block_t block = ^{
```
reverted to the original commit

cyanglaz commented on packages/camera/camera/ios/Classes/QueueUtils.h:
```diff
@@ -7,17 +7,13 @@
-
-/// A class that contains dispatch queue related helper functions.
-@interface QueueHelper : NSObject
+extern const char* FLTCaptureSessionQueueSpecific;
```
nits: Is this used?

hellohuanlin replied:
Yep. This is used to verify if an operation is on capture session queue.

cyanglaz reviewed (approved):
LGTM
